scripts/create-new-package.bash

```diff
@@ -68,30 +69,76 @@ case "$META" in
+
+global_git_name=`git config --global user.name`
+global_git_email=`git config --global user.email`
+if [ -n "$global_git_name" ] && [ -n "$global_git_email" ]; then
+  maintainer_info_global_git="global git: $global_git_name, $global_git_email"
+  maintainer_info_options+=("$maintainer_info_global_git")
 fi
 
-case "$choice" in
-"y")
-  read -p "Enter the maintainer's name: " MAINTAINER_NAME
-  read -p "Enter the maintainer's email address: " MAINTAINER_EMAIL
-  ;;
-"n")
-  MAINTAINER_NAME=`git config user.name`
-  MAINTAINER_EMAIL=`git config user.email`
-  if [[ ! -d ".git" ]]; then
-    cd ..
-    git init >> /dev/null # init git to get data from git command
-    MAINTAINER_NAME=`git config user.name`
-    MAINTAINER_EMAIL=`git config user.email`
-    rm -r .git/
-    cd - >> /dev/null
+local_git_name=""; local_git_email=""
+if [[ -d ".git" ]]; then
+  local_git_name=`git config user.name`
+  local_git_email=`git config user.email`
+  if [ -n "$local_git_name" ] && [ -n "$local_git_email" ]; then
+    maintainer_info_local_git="local git: $local_git_name, $local_git_email"
+    maintainer_info_options+=("$maintainer_info_local_git")
   fi
-esac
+fi
+
+function get_maintainer_name_from_input() {
+  read -p "Enter the maintainer's name: " name
+  while [[ -z $name ]]; do
+      read -p "Name cannot be empty, please enter your name: " name
+  done
+  echo "$name"
+}
+
+function get_maintainer_email_from_input() {
+  read -p "Enter the maintainer's email address: " email
+  while [[ ! $email =~ ^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}$ ]]; do
+      if [[ -z $email ]]; then
+          read -p "Email cannot be empty, please enter your email: " email
+      else
+          read -p "Invalid email format, please enter a valid email: " email
+      fi
+  done
+  echo "$email"
+}
+
+# If there is only user input option ask for maintainer info directly
+if [[ -z "$maintainer_info_global_git" && -z "$maintainer_info_local_git" ]]; then
+  echo "No local or global git name and email found"
+  MAINTAINER_NAME=$(get_maintainer_name_from_input)
+  MAINTAINER_EMAIL=$(get_maintainer_email_from_input)
+else
+  select maintainer_info in "${maintainer_info_options[@]}";
+  do
+    case "$maintainer_info" in
+          "user input")
+              MAINTAINER_NAME=$(get_maintainer_name_from_input)
+              MAINTAINER_EMAIL=$(get_maintainer_email_from_input)
+              break
+            ;;
+          $maintainer_info_global_git)
+              MAINTAINER_NAME=$global_git_name
+              MAINTAINER_EMAIL=$global_git_email
+              break
+            ;;
+          $maintainer_info_local_git)
+              MAINTAINER_NAME=$local_git_name
+              MAINTAINER_EMAIL=$local_git_email
+              break
+            ;;
+    esac
+  done
+fi
+echo -e "${TERMINAL_COLOR_USER_NOTICE}The name '$MAINTAINER_NAME' and email address '$MAINTAINER_EMAIL' will be used as maintainer info!${TERMINAL_COLOR_NC}"
```

Comment:
This is good but it does not enable to write custom name if there is git author specified. Can you also add this. It should be one more switch, similar as before